Build cashlib on travis so that we could run tests based on that on x…

sickpig opened the pull request:
…86_64

brianmcmichael committed:
Qt: Use _putenv_s instead of setenv on Windows builds

dagurval committed:
[rpc] Allow passing int for verbose in getblock

This matches Core more and is required by electrs

Greg-Griffith committed:
Update rpm travis build and doc instructions to use boost 1.66.0

repo is from pkgs.com, it was the only one I could find

sickpig committed:
Partial port of Core's 449f9b8d (BIP141)

Add a helper to determine if CScript is a witness script

sickpig committed:
Add SCRIPT_ALLOW_SEGWIT_RECOVERY

Summary: Flag that allows the spending of p2sh segwit coins

Test Plan: test_bitcoin --run_test=script_tests

Reviewers: deadalnix, O1 Bitcoin ABC, #bitcoin_abc

Reviewed By: deadalnix, O1 Bitcoin ABC, #bitcoin_abc

Subscribers: markblundeberg, teamcity, schancel

Differential Revision: https://reviews.bitcoinabc.org/D2468

floriansengel committed:
Partial port of "Add activation code for SEGWIT_RECOVERY" (D2479)

Test Plan: test_runner.py abc-segwit-recovery-activation

Reviewers: deadalnix, O1 Bitcoin ABC, #bitcoin_abc, jasonbcox, florian

Reviewed By: deadalnix, O1 Bitcoin ABC, #bitcoin_abc, jasonbcox

Subscribers: Fabien, Mengerian, markblundeberg, jasonbcox, teamcity, schancel

Differential Revision: https://reviews.bitcoinabc.org/D2479

sickpig committed:
Add schnorr activation code based on MTP

Remove schnorr activation tweak

sickpig committed:
Build cashlib on travis so that we could run tests based on that on x86_64

sickpig committed:
Build cashlib on travis so that we could run tests based on that on x86_64

markblundeberg committed:
Schnorr signature activation

This is a partial port of https://reviews.bitcoinabc.org/D2483

The functional test need to adapted for BU due to the way ABC
deal with mempool when a reorg happens across a fork block.
Basically they keep in the mempool all the transactions which are
valid on both side of the fork and discard only the ones valid
post-fork.

sickpig commented:
@gandrewstone 

even if we added `--enable-shared` to the configure flags it seems like libbitcoincash.so is not being built. 

Am I doing something wrong? 

```
##################################################
schnorrsig.py:
--------------------------------------------------
stdout:
 Issue loading cashlib shared library.  This is expected during cross compilation since the native python will not load the .so so no error will be reported: /home/travis/build/BitcoinUnlimited/BitcoinUnlimited/src/bitcoind/.libs/libbitcoincash.so: cannot open shared object file: No such file or directory
```

gandrewstone committed:
Merge pull request #1648 from sickpig/SegWit-Recovery

Add SCRIPT_ALLOW_SEGWIT_RECOVERY

gandrewstone commented:
Maybe extra BitcoinUnlimited in the path?

/home/travis/build/***BitcoinUnlimited/BitcoinUnlimited***/src/bitcoind/.libs/libbitcoincash.so

Greg-Griffith committed:
wrap lockedtime in DEBUG_LOCKTIME define

this is a minor performance boost and fixes an inconsistency between CMutexReadLock and CMutexLock

gandrewstone committed:
Merge pull request #1661 from Greg-Griffith/update-rpm-boost-version

Update rpm travis build and doc instructions to use boost 1.66.0

ptschip committed:
Fix spurious failures in mempool_packages.py

Add a syncronization point to wait for the transactions to be
resurrencted in the mempool before proceeding.

ptschip committed:
Temporarily disable maxblocksinflight.py

This is already partially tested in sendheaders.py it would be
good to have a larger test for this, however, it will require a
significant effort for a relatively small reward.

ptschip committed:
Fix spurious failures in bipdersig.py

Run in single threaded mode.

gandrewstone committed:
Fix getblocktemplate_proposals.py for the transaction size limitation

Greg-Griffith committed:
add missing diff for 1589

[skip ci]

gandrewstone committed:
Merge pull request #1666 from ptschip/dev_pytests

Fix spurious failures in mempool_packages.py

gandrewstone committed:
Merge pull request #1667 from gandrewstone/fix_getblocktemplate_proposals_for_gt_100_byte_txs

Fix getblocktemplate_proposals.py for the transaction size limitation

gandrewstone committed:
Merge pull request #1663 from sickpig/schnoor-activation

Schnorr MTP based activation

gandrewstone committed:
Merge pull request #1668 from Greg-Griffith/add-missing-diff-for-1589

add missing diff for 1589

gandrewstone committed:
Merge pull request #1646 from dagurval/getblock-int

[rpc] Allow passing int for verbose in getblock

gandrewstone committed:
Merge pull request #1664 from Greg-Griffith/add-time-define-cmutexreadlock

wrap lockedtime in DEBUG_LOCKTIME define

ptschip committed:
fix bip68-112-113-p2p.py and improve reliability (#1673)

* Move OP_CSV from scriptSig to scriptPubKey in bip68-112-113-p2p

Summary:
Prepare for magnetic anomaly with push-only rule

Depends on D2146

Test Plan:
  ./test/functional/bip68-112-113-p2p.py

Reviewers: #bitcoin_abc, deadalnix, jasonbcox

Reviewed By: #bitcoin_abc, jasonbcox

Subscribers: jasonbcox, teamcity, schancel

Differential Revision: https://reviews.bitcoinabc.org/D2176

* Improve reliablity of bip68-112.113-p2p.py

Greg-Griffith committed:
add get for unsigned ints from univalue class (#1665)

* add get for unsigned ints from univalue class

* add .diff for changes made

bissias committed:
introduce high and low version preferences for graphene (#1670)

gandrewstone committed:
Merge pull request #1675 from ptschip/dev_qt

[Core cherry] Qt: Use _putenv_s instead of setenv on Windows builds

dagurval committed:
[util] Introduce utilprocess (#1645)

* [util] Introduce utilprocess

Currently implements for Linux:
* Send SIGTERM signal to process
* Get current process path

* [util] Implement SubProcess

* [build] Bump minimum supported libc in symbolcheck

Debian Squeeze -> Jessie
Ubuntu Lucid -> Trusty

These are the oldest still supported releases of these distros.

gandrewstone committed:
add schnorr activation test, expand cashlib with datasig, fix lock on xversion (#1672)

ptschip committed:
Refactor Xthin data out of CNode (#1611)

* Move nSizeThinBlock out of CNode

Encapsulate the data into CThinBlock and CXThinBlock

Also encapuslate the thinblock and xthinblock within CBlock. This way
we can eventually just pass around a smart pointer to CBlock and it will
have everything we need to reconstruct the block from the underlying
xthin and/or thinblock data.

* Move thinBlockWaitingForTxns out of CNode

* Move thinblock hashe vectors out of CNode

* fix mapmissing in compactblocks and grapheneblocks

* Move mapMissingTx out of CNode

* Move nLocalThinBlockBytes out of CNode

* Remove CBlock from CNode for thintype relay

Create a subclass from CBlock called CBlock_ThinType and use it

This subclass carries short term data in memory which is used in
the reconstruction of a block from a thinblock or xthinblock.

Rename ClearThinBlockData() to ClearThinBlockBytes()

* Clarify return statement and add a test

* Naming change: CBlock_ThinRelay to CBlockThinRelay

* Use Doxygen commenting for CBlockThinRelay

dagurval committed:
Basic integration with the electrum server electrs (#1633)

This is a MVP for integrating electrum server with Bitcoin Unlimited.

To enable the server, you need to run `make electrs` and add `electrum=1` to `bitcoin.conf`. Adding `debug=electrum` will enable more useful logging.

Greg-Griffith committed:
add unsigned integer types to tweakbase (#1674)

gandrewstone committed:
add understanding of BITCOIND env var to the findBitcoind() function

gandrewstone committed:
add understanding of BITCOIND env var to the findBitcoind() function

sickpig commented:
@gandrewstone fwiw I approve your 3b9e382 changes

gandrewstone committed:
merge

gandrewstone committed:
works locally, not on travis
